### PR TITLE
REV-109 search listings cards

### PR DESCRIPTION
**Pull request by IbrahimAljabrST**
### Jira URL

https://optimumpartners.atlassian.net/browse/REV-109

### Description
-user can see search box at the top of the page
-user is able to insert a city name into search box
-user is able to insert a ZIP code into search box
-User is able to cancel the search by clicking on x button in the search box
-User is able to submit the search by pressing “enter”
-User is able to submit the search by clicking on search icon
-User is able to see number of results after proceed with the search

### Dependencies in Other PRs,Repo

https://github.com/abukhadraop-or/reverifi-api/pull/24

### CODE REVIEW

File: src/components/home/Storyboard/index.jsx [hunk 1]
```diff
@@ -20,21 +23,23 @@ import {
...
-      const appPaths = {
-        listingPaths: {
-          search: '/listing/search',
-        },
-      };
-
-      navigate(`${appPaths.listingPaths.search}?key=${address}`);
+    if (inputValue.current.value) {
+      navigate(
+        `${appPaths.listingPaths.search}?key=${inputValue.current.value}`
```
Comment (AhmadAlShwaiki) [inline src/components/home/Storyboard/index.jsx:33]: destruct please
```suggestion
        `${listingPaths.search}?key=${inputValue.current.value}`
```

File: src/components/listing-search-page/SearchResults/index.jsx [hunk 2]
```diff
@@ -41,17 +51,32 @@ function ListingSearch() {
...
+        <InputWrapper>
+          <input
+            ref={inputValue}
+            type="text"
+            defaultValue={keyWord}
+            onChange={(e) => {
+              setKeyWord(e.target.value);
+            }}
+            onKeyDown={(e) => {
+              if (e.key === 'Enter') {
```
Comment (AhmadAlShwaiki) [inline src/components/listing-search-page/SearchResults/index.jsx:72]: Please use [keyCodes](https://css-tricks.com/snippets/javascript/javascript-keycodes/)

File: src/components/listing-search-page/SearchResults/listing-search-cards.style.js [hunk 3]
```diff
@@ -37,14 +39,58 @@ export const SearchButton = styled.button`
 
 export const CardsContainer = styled.div`
   max-height: 40rem;
-  overflow: scroll;
+  overflow-y: scroll;
+`;
+
+export const SorryMessage = styled.div`
```
Comment (AhmadAlShwaiki) [inline src/components/listing-search-page/SearchResults/listing-search-cards.style.js:45]:
```suggestion
export const Container = styled.div`
```
To hold both messages

File: src/components/listing-search-page/SearchResults/listing-search-cards.style.js [hunk 4]
```diff
@@ -58,8 +104,21 @@ export const SearchContainer = styled.div`
      `}
   }
 
-  button {
-    margin: 0 0.7rem;
-    padding: 0.3rem 0.8rem;
+  svg {
```
Comment (AhmadAlShwaiki) [inline src/components/listing-search-page/SearchResults/listing-search-cards.style.js:107]: Please create a styled component

File: src/components/listing-search-page/SearchResults/listing-search-cards.style.js [hunk 5]
```diff
@@ -37,14 +39,58 @@ export const SearchButton = styled.button`
...
+`;
+
+export const ResultNumber = styled.p`
+  font-weight: 600;
+  padding-right: 0.5rem;
+`;
+
+export const ResultText = styled.p`
+  font-weight: 600;
+  color: ${colors.dustyGray};
```
Comment (AhmadAlShwaiki) [inline src/components/listing-search-page/SearchResults/listing-search-cards.style.js:77]:
```suggestion
export const Text = styled.p`
....
```
Please create one shared Text then extend your components

File: src/components/listing-search-page/Card/card.style.js [hunk 6]
```diff
@@ -30,21 +34,23 @@ export const ImageContainer = styled.div`
...
+    max-width: 10rem;
   `}
+
+  ${mq.desktop`
+      max-height: 20rem;
+      max-width: 10rem;
+    `}
 `;
 
 export const Badge = styled.div`
```
Comment (mohdabed123) [inline src/components/listing-search-page/Card/card.style.js:46]: why are we using a div instead of heading

**Review by mohdabed123 — CHANGES_REQUESTED** [review]

File: src/components/listing-search-page/SearchResults/index.jsx [hunk 7]
```diff
@@ -65,11 +90,26 @@ function ListingSearch() {
         </Button>
       </SearchContainer>
 
-      <CardsContainer>
-        {cardData?.map((item) => (
-          <Card data={item} key={item.id} />
-        ))}
-      </CardsContainer>
+      {cardData?.length ? (
```
[the remark this answers is not in the document]
Comment (IbrahimAljabrST) [reply src/components/listing-search-page/SearchResults/index.jsx:93]: empty array will give true

File: src/components/home/Storyboard/index.jsx [hunk 8]
```diff
@@ -20,21 +23,21 @@ import {
...
-      };
-
-      navigate(`${appPaths.listingPaths.search}?key=${address}`);
+    if (inputValue.current.value) {
+      navigate(`${listingPaths.search}?key=${inputValue.current.value}`);
     }
   };
 
+  const handleInputClear = () => {
+    inputValue.current.value = '';
```
Comment (AhmadAlShwaiki) [inline src/components/home/Storyboard/index.jsx:37]:
```suggestion
    setCheckValue```

File: src/components/home/Storyboard/index.jsx [hunk 9]
```diff
@@ -20,21 +23,21 @@ import {
  * @return {JSX.Element}
  */
 function Storyboard() {
-  const [address, setAddress] = useState(null);
   const navigate = useNavigate();
+  const [checkValue, setCheckValue] = useState(null);
```
Comment (AhmadAlShwaiki) [inline src/components/home/Storyboard/index.jsx:27]:
```suggestion
  const [address, setAddress] = useState(null);
```

File: src/components/listing-search-page/Card/index.jsx [hunk 10]
```diff
@@ -28,39 +37,111 @@ import {
...
-            <BathtubIcon />
-          </IconsNumber>
-          <AirConditionerIcon />
-          <BenchIcon />
+          {data.bedrooms ? (
+            <IconsNumber>
+              {data.bedrooms}
+              <Bed />
+            </IconsNumber>
+          ) : null}
```
Comment (AhmadAlShwaiki) [inline src/components/listing-search-page/Card/index.jsx:74]: Same for all other elements please 
```suggestion
          {bedrooms && (
            <IconsNumber>
              {data.bedrooms}
              <Bed />
            </IconsNumber>
          ) }
```

File: src/components/listing-search-page/SearchResults/index.jsx [hunk 11]
```diff
@@ -22,54 +31,88 @@ import {
 function ListingSearch() {
   const [keyWord, setKeyWord] = useState(window.location.href.split('=')[1]);
```
Comment (AhmadAlShwaiki) [inline src/components/listing-search-page/SearchResults/index.jsx:32]:
```suggestion
useLocation()
```

File: src/components/listing-search-page/SearchResults/index.jsx [hunk 12]
```diff
@@ -22,54 +31,88 @@ import {
...
   const [cardData, setCardData] = useState(null);
+  const [resultCount, setResultCount] = useState(0);
```
Comment (AhmadAlShwaiki) [inline src/components/listing-search-page/SearchResults/index.jsx:34]:
```suggestion
```

File: src/components/listing-search-page/SearchResults/index.jsx [hunk 13]
```diff
@@ -22,54 +31,88 @@ import {
...
-        {cardData?.map((item) => (
-          <Card data={item} key={item.id} />
-        ))}
-      </CardsContainer>
+      {cardData?.length ? (
+        <>
+          <ResultCount>
+            <ResultNumber>{resultCount}</ResultNumber>
+            <ResultText>{keyWord}</ResultText>
+          </ResultCount>
```
Comment (AhmadAlShwaiki) [inline src/components/listing-search-page/SearchResults/index.jsx:100]: This component change in satet
```suggestion
          <ResultCountContainer>
            <ResultNumber>{results.?length ? results.?length: 0 }</ResultNumber>
            <ResultText>{keyWord}</ResultText>
          </ResultCountContainer>
```
Can you please explain to me why are you using state for component management here?

File: src/components/listing-search-page/SearchResults/index.jsx [hunk 14]
```diff
@@ -22,54 +31,88 @@ import {
...
+            ))}
+          </CardsContainer>
+        </>
+      ) : (
+        <>
+          <MessageContainer>
+            <Message>{sorryMessage}</Message>
+          </MessageContainer>
+          <SuggestedListing>Suggested Listings you may like</SuggestedListing>
+          {/* todo */}
```
Comment (AhmadAlShwaiki) [inline src/components/listing-search-page/SearchResults/index.jsx:113]:
```suggestion
Please remove this, you can manage all actions and items to add in Jira tickets
```

**Review by AhmadAlShwaiki — APPROVED** [review]

**Review by mohdabed123 — APPROVED** [review]

**Pull request merged**